tuned_lens/model_surgery.py

```diff
@@ -221,6 +221,87 @@ def replace_layers(
 
     set_key_path_(model, list_path, modified_list)
     try:
-        yield model
+        yield
     finally:
         set_key_path_(model, list_path, layer_list)
+
+
+class _AddVector(th.nn.Module):
+    def __init__(
+        self,
+        base: th.nn.Module,
+        vector: th.Tensor,
+        get_hidden: Callable[[Any], th.Tensor],
+        set_hidden: Callable[[Any, th.Tensor], Any],
+    ):
+        super().__init__()
+        self.base = base
+        self.vector = nn.Parameter(vector.clone())
```

Comment:
Is this intended to allow optimizers to optimize the steering vector?
- If yes, how can optimizers list the parameters? (upon reading more of the code, it's possible by calling `model.parameters()` inside the context.
- if not, this should be `register_buffer()` or have the arg `requires_grad=False`

Reply:
Okay I will change this